### ISSUE TITLE
Add Mcdropout layer

### ISSUE THREAD

**Issue by oskarpfeffer** [pytorch/pytorch#136994]
### 🚀 The feature, motivation and pitch

I am working with MCDropout, and find the current workflow in pytorch suboptimal and misleading. The current way to enable Dropout at evaluation time to perform MCDropout (https://arxiv.org/pdf/1506.02142) is to set all the Dropout Layers to training mode, which is in my opinion misleading, since you are not training at that point. 
My implementation would be to set a second flag `mcdropout` that enables to use the Layer with dropout always turned on, so that one does not need to loop over the layers, maintaining a clearer and cleaner code structure.

This is my suggestion:
```
class MCDropout(_DropoutNd):
    """
    Implementation of MCDropout to the torch dropout layer.
    This adds a mcdropout flag, which turns dropout always on in training and
    evaluation mode.

    Args:
        p : probability of an element to be zeroed. Default: 0.5
        mcdropout: if True, will always perform dropout
        inplace : if True, will perform dropout in-place
    """
    def __init__(
        self, p: float = 0.5, mcdropout: bool = True, inplace: bool = False
    ) -> None:
        super().__init__(p, inplace)
        self.mcdroput = mcdropout

    def forward(self, input: torch.Tensor) -> torch.Tensor:
        return torch.nn.functional.dropout(
            input, self.p, self.training or self.mcdropout, self.inplace
        )
```

The change adds a parameter `mcdropout` to the Dropout layer, and returns as a condition to use dropout or not `self.training or self.mcdropout`, thus avoiding the need to go over all the layers and turn the training mode on. This also enables to set only some layers to MCDropout layers in an easier way.

### Alternatives

Currently to use mcdropout one has to go over all the modules in the model and set them to training mode.
```def enable_dropout(model):
    """ Function to enable the dropout layers during test-time """
    for m in model.modules():
        if m.__class__.__name__.startswith('Dropout'):
            m.train()
```
